packages/medusa-plugin-auth/src/auth-strategies/auth0/store.ts

```diff
@@ -0,0 +1,161 @@
+import passport from 'passport';
+import { Router } from 'express';
+import cors from 'cors';
+import { ConfigModule, MedusaContainer } from '@medusajs/medusa/dist/types/global';
+import jwt from 'jsonwebtoken';
+import { Strategy as Auth0Strategy } from 'passport-auth0';
+import { CustomerService } from '@medusajs/medusa';
+import { MedusaError } from 'medusa-core-utils';
+import { EntityManager } from 'typeorm';
+import {
+	CUSTOMER_METADATA_KEY,
+	AUTH_PROVIDER_KEY,
+	STORE_AUTH_TOKEN_COOKIE_NAME,
+	TWENTY_FOUR_HOURS_IN_MS,
+} from '../../types';
+import { buildCallbackHandler } from '../../core/utils/build-callback-handler';
+import { PassportStrategy } from '../../core/Strategy';
+import { Auth0Options, Profile, ExtraParams, AUTH0_STORE_STRATEGY_NAME } from './types';
+
+export class Auth0StoreStrategy extends PassportStrategy(Auth0Strategy, AUTH0_STORE_STRATEGY_NAME) {
+	constructor(
+		protected readonly container: MedusaContainer,
+		protected readonly configModule: ConfigModule,
+		protected readonly strategyOptions: Auth0Options
+	) {
+		super({
+			domain: strategyOptions.auth0Domain,
+			clientID: strategyOptions.clientID,
+			clientSecret: strategyOptions.clientSecret,
+			callbackURL: strategyOptions.store.callbackUrl,
+			passReqToCallback: true,
+			state: true,
+		});
+	}
+
+	async validate(
+		req: Request,
+		accessToken: string,
+		refreshToken: string,
+		extraParams: ExtraParams,
+		profile: Profile
+	): Promise<null | { id: string }> {
+		if (this.strategyOptions.store.verifyCallback) {
+			return await this.strategyOptions.store.verifyCallback(
+				this.container,
+				req,
+				accessToken,
+				refreshToken,
+				extraParams,
+				profile
+			);
+		}
+		return await this.defaultValidate(profile);
+	}
+
+	private async defaultValidate(profile: Profile): Promise<{ id: string } | never> {
+		const manager: EntityManager = this.container.resolve('manager');
+		const customerService: CustomerService = this.container.resolve('customerService');
+
+		return await manager.transaction(async (transactionManager) => {
+			const email = profile.emails?.[0]?.value;
+
+			if (!email) {
+				throw new MedusaError(
+					MedusaError.Types.NOT_ALLOWED,
+					`Your Auth0 account does not contain a valid email and cannot be used`
+				);
+			}
+
+			const customer = await customerService
```

Comment:
in the newest version of medusa, the retrieval of the customer has changed a bit. We can have now multiple customer with the same email. One as a guest and one with an account. I suggest that we check if `retrieveRegisteredByEmail` exists on the customer service and fallback to `retrieveByEmail` otherwise. wdyt? so that we can support older and newer versions of medusa

Reply:
I see you have refactored to only do `retrieveRegisteredByEmail`, should we still add the fallback?